### PR TITLE
Socint 117

### PR DESCRIPTION
**Pull request by KieranWardle**
Added new events to Event Publisher

### CODE REVIEW

**Review by spdiaz1000 — APPROVED** [review]

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/EventTopic.java [hunk 1]
```diff
@@ -9,7 +9,10 @@
   SURVEY_LAUNCH("event_survey-launch", EventType.SURVEY_LAUNCH),
   REFUSAL("event_refusal", EventType.REFUSAL),
   UAC_UPDATE("event_uac-update", EventType.UAC_UPDATE),
-  CASE_UPDATE("event_case-update", EventType.CASE_UPDATE);
+  CASE_UPDATE("event_case-update", EventType.CASE_UPDATE),
+  SURVEY_UPDATE("event_survey-update", EventType.SURVEY_UPDATE),
+  COLLECTION_EXERCISE_UPDATE(
+      "event_collection-exercise-update", EventType.COLLECTION_EXERCISE_UPDATE); // TODO topic name
```
Comment (rpcatling) [inline event-publisher/src/main/java/uk/gov/ons/ctp/common/event/EventTopic.java:15]: please correct the topic name and remove the TODO comment.

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import uk.gov.ons.ctp.common.jackson.CustomDateSerialiser;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class CollectionExercise implements EventPayload {
+  private String collectionExerciseId;
+  private String surveyId;
+  private String name;
+  private String reference;
```
Comment (rpcatling) [inline event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java:17]: where is the requirement for `reference` ?

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  @JsonSerialize(using = CustomDateSerialiser.class)
```
Comment (rpcatling) [inline event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java:19]: where is the requirement for the start/end and metadata ?

**Review by rpcatling — CHANGES_REQUESTED** [review]
Please merge with `main`

Also need unit tests : add Json test files for new events and add unit tests to `EventPublisherTest`

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/EventTopic.java (hunk 1 above)
Comment (KieranWardle) [reply event-publisher/src/main/java/uk/gov/ons/ctp/common/event/EventTopic.java]: Topic name was already correct, TODO removed

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java (hunk 2 above)
Comment (KieranWardle) [reply event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java:17]: https://github.com/ONSdigital/ssdc-shared-events/blob/main/event_dictionary/v0.3_RELEASE/collectionExerciseUpdate.schema.json

I got the event definition from here, it contains `reference` and the fields mentioned in the comment below

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java (hunk 3 above)
Comment (KieranWardle) [reply event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java:19]: Merged with main and added unit tests

**Comment by KieranWardle** [comment]
Merged with main and added unit tests

File: event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java (hunk 2 above)
Comment (rpcatling) [reply event-publisher/src/main/java/uk/gov/ons/ctp/common/event/model/CollectionExercise.java:17]: OK , it has changed from the DRAFT to the RELEASE

**Review by rpcatling — APPROVED** [review]

**Pull request merged**